cylc/flow/subprocpool.py

```diff
@@ -487,7 +488,14 @@ def _run_callback(callback, args_=None):
         # that you can use that platform's ssh command.
         platform = None
         if isinstance(ctx.cmd_key, TaskJobLogsRetrieveContext):
-            platform = get_platform(ctx.cmd_key.platform_name)
+            try:
```

Comment:
I've kept this one reasonably simple. I've struggled to recreate this error by changing externals and in the end was only able to manually test it by adding

```python
               raise PLATFORM_LOOKUP_ERRORS[2]('foobar')
```

This isn't really very satisfactory, but I don't think that this likely to cause a real world bug, and that this case is just covering ourselves.